main/lsp/LSPTypecheckerCoordinator.cc

```diff
@@ -129,14 +131,18 @@ void LSPTypecheckerCoordinator::syncRun(unique_ptr<LSPTask> task) {
     wrappedTask->blockUntilComplete();
 }
 
-void LSPTypecheckerCoordinator::initialize(unique_ptr<InitializedParams> params) {
-    // TODO: Split into a sync and async task when we land preemptible slow path, where the typecheck is async.
-    syncRun(make_unique<InitializeTask>(*config, typechecker, workers, move(params->updates)));
+void LSPTypecheckerCoordinator::initialize(LSPFileUpdates initialUpdate) {
+    // TODO: Make an async task when we land preemptible slow path, where the typecheck is async.
```

Comment:
It doesn't make sense to make initialization asynchronous now, as the next message would end up blocked anyway.

Reply:
>  Or by "now," do you mean, "it currently doesn't make sense, but it will in the future"?

It currently doesn't make sense. It will make sense in the future.

> In that case, don't you still want the old comment, because there will always be a synchronous part?

Nope! `InitializeTask` itself will be asynchronous. I was mistaken when we chatted before; `InitializeTask` is basically a specialized run slow path. The synchronous portion is the indexing and hashing task in request_dispatch, which previously ran in the preprocessor.